core/src/main/scala/sqlitis/Parser.scala

```diff
@@ -133,14 +131,14 @@ object Parser {
         }
     )
 
-  implicit lazy val select: P[Select[Unit]] = (
-    kw("SELECT") *> (kw("DISTINCT")).?.map(_.isDefined) ~
-      repSep(field, 0, comma) ~
-      optKw("FROM", repSep(from, 1, comma)).map(_.getOrElse(Nil)) ~
-      optKw("WHERE", expr <* maybeWhitespace) ~
-      optKw("ORDER BY", orderBy) ~
-      optKw("LIMIT", token(int))
-  ).map { case (((((isDistinct, fields), from), where), orderBy), limit) =>
+  implicit val select: P[Select[Unit]] = (
+    kw("SELECT") *> (kw("DISTINCT")).?.map(_.isDefined),
```

Comment:
I think you wanted `kw("SELECT")` outside of the `(...` which would then make the resulting parser a Parser1, which can be useful sometimes...